### PR TITLE
[docs] Add owner for darwin/amd64

### LINKED ISSUE #9114: Find owners for the darwin/amd64 platform

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Adds owner for darwin/amd64 platform based on https://github.com/open-telemetry/opentelemetry-collector/issues/8526#issuecomment-1905979165.

**Link to tracking Issue:** Fixes #9114

### CODE REVIEW

**Comment by mx-psi** [comment]
@h0cheung can you review?

File: docs/platform-support.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ Tier 3 platforms are _guaranteed to build_. Precompiled binaries are made availa
 Tier 3 platforms are currently:
 | Platform      | Owner(s)                                                                                                    |
 |---------------|-------------------------------------------------------------------------------------------------------------|
-| darwin/amd64  |                                                                                                             |
+| darwin/amd64  | [h0cheung](https://github.com/h0cheung)                                                                     |
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply docs/platform-support.md]: Thanks!

File: docs/platform-support.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ Tier 3 platforms are _guaranteed to build_. Precompiled binaries are made availa
 Tier 3 platforms are currently:
 | Platform      | Owner(s)                                                                                                    |
 |---------------|-------------------------------------------------------------------------------------------------------------|
-| darwin/amd64  |                                                                                                             |
+| darwin/amd64  | [@h0cheung](https://github.com/h0cheung)                                                                     |
```
Comment (mx-psi) [inline docs/platform-support.md:61]:
```suggestion
| darwin/amd64  | [@h0cheung](https://github.com/h0cheung)                                                                    |
```

**Review by codeboten — APPROVED** [review]
Thanks @h0cheung!

**Pull request merged**